lib/Runtime/Library/SharedArrayBuffer.cpp

```diff
@@ -246,9 +246,38 @@ namespace Js
         return JavascriptOperators::GetTypeId(aValue) == TypeIds_SharedArrayBuffer;
     }
 
-    SharedArrayBuffer::SharedArrayBuffer(uint32 length, DynamicType * type) :
-        ArrayBufferBase(type), sharedContents(nullptr)
+    BYTE* SharedArrayBuffer::AllocBuffer(uint32 length, uint32 maxLength)
+    {
+#if ENABLE_FAST_ARRAYBUFFER
+        if (this->IsValidVirtualBufferLength(length))
+        {
+            return (BYTE*)AsmJsVirtualAllocator(length);
+        }
+        else
+#endif
+        {
+            return HeapNewNoThrowArray(BYTE, length);
+        }
+    }
+
+    void SharedArrayBuffer::FreeBuffer(BYTE* buffer, uint32 length, uint32 maxLength)
```

Comment:
nit: there deserve to be some comments explaining maxLength, since this is a wasm specific parameter, and makes no sense for JS where you can't grow SABs